blockchain/vmQueryGetter.go

```diff
@@ -0,0 +1,170 @@
+package blockchain
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+	"strconv"
+
+	"github.com/multiversx/mx-chain-core-go/core/check"
+	logger "github.com/multiversx/mx-chain-logger-go"
+	"github.com/multiversx/mx-sdk-go/builders"
+	"github.com/multiversx/mx-sdk-go/data"
+)
+
+const (
+	okCodeAfterExecution = "ok"
+	internalError        = "internal error"
+	vmQuerylogPath       = "blockchain/vmQueryGetter"
+)
+
+type vmQueryGetter struct {
+	proxy Proxy
+	log   logger.Logger
+}
+
+// NewVmQueryGetter creates a new instance of the vmQueryGetter type
+func NewVmQueryGetter(proxy Proxy) (*vmQueryGetter, error) {
+	if check.IfNil(proxy) {
+		return nil, ErrNilProxy
+	}
+
+	client := vmQueryGetter{
+		proxy: proxy,
+	}
+	client.log = logger.GetOrCreate(vmQuerylogPath)
+	return &client, nil
+}
+
+// ExecuteQueryReturningBytes will try to execute the provided query and return the result as slice of byte slices
+func (dataGetter *vmQueryGetter) ExecuteQueryReturningBytes(ctx context.Context, request *data.VmValueRequest) ([][]byte, error) {
+	if request == nil {
+		return nil, ErrNilRequest
+	}
+
+	response, err := dataGetter.proxy.ExecuteVMQuery(ctx, request)
+	if err != nil {
+		dataGetter.log.Error("got error on VMQuery", "FuncName", request.FuncName,
```

Comment:
let's just return a wrapped error instead of log + return error. This will prevent double/multiple logging of the same error

Reply:
done